### PR TITLE
⚠️ Bump to controller-runtime v0.16

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:
I'll start verifying controller-runtime main on this PR.

Not really worth reviewing until we're closer to the Kubernetes / CR release (at some point towards end of August)

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Part of #8708

### CODE REVIEW

**Review by killianmuldoon — COMMENTED** [review]
/lgtm
/approve

/hold
(For CI and in case anyone else wants to take a look)

**Review by furkatgofurov7 — COMMENTED** [review]
/lgtm

File: docs/book/src/developer/providers/migrations/v1.5-to-v1.6.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ maintainers of providers and consumers of our Go API.
 **Note**: Only the most relevant dependencies are listed, `k8s.io/` and `ginkgo`/`gomega` dependencies in Cluster API are kept in sync with the versions used by `sigs.k8s.io/controller-runtime`.
 
 - sigs.k8s.io/kind: v0.20.x
-- sigs.k8s.io/controller-runtime: v0.15.x
+- sigs.k8s.io/controller-runtime: v0.15.x => v0.16.x
```
Comment (furkatgofurov7) [inline docs/book/src/developer/providers/migrations/v1.5-to-v1.6.md:15]: should we just say:

```suggestion
- sigs.k8s.io/controller-runtime: v0.16.x
```

instead?

Comment (sbueringer) [reply docs/book/src/developer/providers/migrations/v1.5-to-v1.6.md:15]: To be honest, I'm fine either way

Comment (furkatgofurov7) [reply docs/book/src/developer/providers/migrations/v1.5-to-v1.6.md:15]: Fine as is then.

**Comment by sbueringer** [comment]
Moved the fix for kubekins/kubectl to a separate PR: https://github.com/kubernetes-sigs/cluster-api/pull/9275

This PR now depends on #9275 being merged

**Review by furkatgofurov7 — COMMENTED** [review]
/lgtm

File: bootstrap/kubeadm/main.go [hunk 2]
```diff
@@ -218,9 +220,12 @@ func main() {
 		LeaderElectionResourceLock: resourcelock.LeasesResourceLock,
 		HealthProbeBindAddress:     healthAddr,
 		PprofBindAddress:           profilerAddress,
+		Metrics: metricsserver.Options{
+			BindAddress: metricsBindAddr,
```
Comment (fabriziopandini) [inline bootstrap/kubeadm/main.go:224]: q: as a follow-up, what about opening an issue for documenting how to get metrics in CAPI?

**Review by fabriziopandini — COMMENTED** [review]
/lgtm
/approve

great work!
Nice to see how this bump went in smoothly

File: bootstrap/kubeadm/main.go (hunk 2 above)
Comment (sbueringer) [reply bootstrap/kubeadm/main.go:224]: In this PR nothing changes.

Authorized metrics will be part of this effort: https://github.com/kubernetes-sigs/cluster-api/pull/9264
I was planning to write a corresponding issue and bring it up in the office hours.

But just the CR bump itself in this PR doesn't change anything

**Review by furkatgofurov7 — COMMENTED** [review]
/lgtm

**Comment by sbueringer** [comment]
/hold cancel 

Given all the lgtm's above

**Pull request merged**